R/SEIaImIsRD.R

```diff
@@ -0,0 +1,294 @@
+#' @export SEIaImIsRD
+SEIaImIsRD <- setClass(Class = "SEIaImIsRD",
+         slots = c(
+           initial_condition_names = "list",
+           transmission_parameter_names = "list",
+           initial_conditions = "list",
+           transmission_parameters = "list",
+           R0 = "numeric",
+           output_names = "list",
+           output = "list"
+         ),
+         prototype = list(
+           initial_condition_names = list("S0", "E0", "I_asymptomatic0",
+                                          "I_mild0", "I_severe0", "R0", "D0"),
+           transmission_parameter_names = list("beta", "kappa", "omega",
+                                              "p_symptom", "gamma", "mu"),
+           initial_conditions = vector(mode = "list", length = 6),
+           transmission_parameters = vector(mode = "list", length = 6),
+           R0 = NA_real_,
+           output_names = list("S", "E", "I_asymptomatic",
+                               "I_mild", "I_severe", "R", "D"),
+           output = data.frame()
+         ))
+
+
+
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Functions
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+
+#' set initial population sizes (in fraction) and parameters
+#'
+#' @param object An object of class SEIaImIsRD
+#' @param S Numeric, initial fraction of the population that is susceptible
+#' @param E Numeric, initial fraction of the population that is exposed
+#' @param I_asymptomatic Numeric, initial fraction of the population that is infected with no symptom
+#' @param I_mild Numeric, initial fraction of the population that is infected with mild symptoms
+#' @param I_severe Numeric, initial fraction of the population that is infected severe symptoms
+#' that need further hospitalization
+#' @param D Numeric, initial fraction of the population that is dead due to the infection
+#' @param R Numeric, initial fraction of the population that is recovered
+#' @param beta Numeric vector, effective contact rate (i.e. rate at which an infected individual exposes susceptible) from different symptome compartments
+#' @param kappa Numeric, rate of progression from exposed to infectious (the reciprocal is the incubation period).
+#' @param omega Numeric, rate at which recovered individuals become susceptible
+#' @param p_symptom Numeric vector, fractions of different symptome compartments
+#' @param gamma Numeric vector, rate of removal (i.e. rate at which each infected group recover).
+#' @param mu Numeric vector, rate of disease-caused mortality of each infected group
+#' @return An object of class SEIaImIsRD with initial population and parameters
+#' @rdname SEIaImIsRD-class
+#' @export
+#'
+set_init <- function(
+  object,
+  S=NA_real_,
+  E=NA_real_,
+  I_asymptomatic=NA_real_, I_mild=NA_real_, I_severe=NA_real_,
+  R=NA_real_,
+  D=NA_real_,
+  beta=list(), kappa=NA_real_, omega=NA_real_,
+  p_symptom=list(), gamma=list(), mu=list()
```

Comment:
It's not obvious how the named list of `p_symptom, gamma` and `mu` should be constructed for the function input. It might be better to show in some way what names to use for the lists.